tuf/api/metadata.py

```diff
@@ -1123,6 +1124,41 @@ def to_dict(self) -> Dict[str, Any]:
             **self.unrecognized_fields,
         }
 
+    @classmethod
+    def from_file(cls, filepath: str) -> "TargetFile":
+        """Creates TargetFile object from a file."""
+        # Ensure the file path is valid/exists.
+        if not os.path.exists(filepath):
+            raise exceptions.NotFoundError(repr(filepath) + " does not exist.")
+
+        with open(filepath, "rb") as file:
+            data = file.read()
+
+        return cls.from_data(data)
```

Comment:
we need to think about the case of very large files here (probably not uncommon for target files), and not read to memory: your from_data() looks complete enough that I _think_ you can just do 
```suggestion
        with open(filepath, "rb") as file:
            return cls.from_data(file)
```
this should end up in the file object branch which uses digest_fileobject() which does not read the whole file at once.

Reply:
Thanks, @jku! Addressed.